### PR TITLE
Improved glossary and let plugins add items to it

### PR DESCRIPTION
**Pull request by tsteur**
The current glossary is pretty plain and a bit hard to read and does not even have category titles like "Reports" or "Metrics" see https://glossary.matomo.org/

As I needed to generalize the glossary a bit more so plugins can modify the glossary, I tweaked it a bit using Materialize components tabs, scrollspy and pagination. 

image

Might need to update the changelog if we put this into 3.6.0. I presume for 3.5.1 we don't have enough time.

### CODE REVIEW

**Comment by tsteur** [comment]
Failing UI tests do not seem to be related to this change.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ The Product Changelog at **[matomo.org/changelog](https://matomo.org/changelog)*
 ### New APIs
 
 * Added new method `Piwik\API\Request::isRootRequestApiRequest()` to detect if the root request is an API request.
+* Added new event `API.addGlossaryItems` which lets you add items to the glossary.
```
Comment (sgiehl) [inline CHANGELOG.md]: Needs to be moved to the correct place before merging

**Review by sgiehl — APPROVED** [review]
Looks good so far. We could maybe add a UI test for the glossary

**Comment by diosmosis** [comment]
The glossary looks great 👍

**Pull request merged**